controllers/storagecluster/external_ocs.go

```diff
@@ -107,9 +116,64 @@ func (r *StorageClusterReconciler) acknowledgeOnboarding(instance *ocsv1.Storage
 	instance.Status.Phase = statusutil.PhaseProgressing
 
 	r.Log.Info("External-OCS:Onboarding is acknowledged successfully.")
+
 	return reconcile.Result{Requeue: true}, nil
 }
 
+func (r *StorageClusterReconciler) reconcileConsumerStatusReporterJob(instance *ocsv1.StorageCluster, externalClusterClient *externalClient.OCSProviderClient) (reconcile.Result, error) {
+	// start the cronJob to ping the provider api server
+	cronJob := &batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "report-status-to-provider",
+			Namespace: instance.Namespace,
+		},
+	}
+
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.Client, cronJob, func() error {
+		cronJob.Spec = batchv1.CronJobSpec{
+			Schedule: "* * * * *",
+			JobTemplate: batchv1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "heartbeat",
+									Image: os.Getenv(providerAPIServerImage),
+									Command: []string{
+										"/usr/local/bin/status-reporter",
+									},
+									Env: []corev1.EnvVar{
+										{
+											Name:  StorageClusterNamespaceEnvVar,
+											Value: instance.Namespace,
+										},
+										{
+											Name:  StorageClusterNameEnvVar,
+											Value: instance.Name,
+										},
+									},
+								},
+							},
+							RestartPolicy:      corev1.RestartPolicyOnFailure,
+							ServiceAccountName: "ocs-status-reporter",
+						},
+					},
+				},
+			},
+		}
+		if err := controllerutil.SetOwnerReference(instance, cronJob, r.Client.Scheme()); err != nil {
+			return fmt.Errorf("failed to set owner reference: %v", err)
+		}
```

Comment:
Can we please move this above the spec update?

Reply:
Updated in the latest commit